keps/1397-evictions-in-background/README.md

```diff
@@ -0,0 +1,1147 @@
+- `sigs.k8s.io/descheduler/pkg/framework/plugins/removepodsviolatingtopologyspreadconstraint`: `2024-05-13` - `43.0`
+
+These tests will be added:
+- New tests  will be added under `sigs.k8s.io/descheduler/pkg/framework/plugins` for each
+  mentioned plugin to test evacuation (replacement of the eviction API).
+- New tests will be added under `sigs.k8s.io/descheduler/pkg/descheduler/evictions` to test
+  the new evacuation (annotation based, API based) implementation.
+- New tests under  `sigs.k8s.io/descheduler/pkg/descheduler` will be created for:
+  - the new internal sorting and filtering algorithms
+  - validating the evacuation limits are respected with already existing
+    evacuation requests from a previous descheduling cycle
+
+##### Integration tests
+
+<!--
+Integration tests are contained in k8s.io/kubernetes/test/integration.
+Integration tests allow control of the configuration parameters used to start the binaries under test.
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+No integration tests available
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+These tests will be added:
+- `EvacuationInBackground` for testing an evacuation request/annotated pod eviction:
+  1. Setup a validating admission webhook that will intercept an eviction request,
+     start a pod eviction/deletion in e.g. T=5s but rejects the eviction request.
+  2. Create an evacuation request for a pod/evict a pod
+     with `descheduler.alpha.kubernetes.io/request-evict-only` annotation.
+  3. Observe the pod is not removed right away and wait for the configured
+     delay T before checking a pod is removed.
+  4. Check a pod was removed after T, not before.
+- `EvacuationLimitReached` for testing only a limited amount of pods is evacuated
+  with already existing evacuation requests:
+  1. Create evacuation requests (e.g. by annotating pods
+     with `descheduler.alpha.kubernetes.io/eviction-in-progress` or creating evacuation requests).
+  2. Run the descheduler and observe only an expected amount of pods
+     is requested to be evicted/evacuated.
+- `EvacuationTTL` for testing an evacuation request (with a single owner
```

Comment:
I am a bit confused, do we introduce the TTL for the first version or not?

Reply:
I see I have mentioned the TTL will not be implemented in the first version under Open Questions. Yet, v1alpha1 section mentions it will. I removed that part from v1alpha1.

v1alpha1 will not introduce the TTL mechanism. Yet, it might get implemented in later v1alpha versions.